sema/check_function.go

```diff
@@ -455,6 +456,8 @@ func (checker *Checker) visitWithPostConditions(
 				resultType = returnType
 			}
 
+			checker.Elaboration.SetResultVariableType(enclosingElement, resultType)
```

Comment:
Can we just use the return type of the function type instead?

Reply:
Unfortunately no, for resources, the result variable's type becomes a reference.
We could theoretically re-calculate the type at the compiler (i.e: if it's a resource, then use the reference type), but it would be less error-prone to do the calculation once (which is already done at the checker), and then cache and re-use it

Reply:
Ah, I see! In that case it makes sense to store it 👍